main/main.cpp

```diff
@@ -329,3 +357,18 @@ int main(void)
     NRF_LOG_INFO("vTaskStartScheduler() failed");
     APP_ERROR_HANDLER(0);
 }
+
+/* Handle events from the Weave Device layer.
+ *
+ * NOTE: This function runs on the Weave event loop task.
+ */
+void DeviceEventHandler(const WeaveDeviceEvent * event, intptr_t arg)
+{
+    if (event->Type == DeviceEventType::kSessionEstablished &&
+        event->SessionEstablished.IsCommissioner)
+    {
+        // Disable advertising when Commissioner connected over WoBLE
+        // so ToBLE connectivity checks can occur.
+        ConnectivityMgr().SetBLEAdvertisingEnabled(false);
```

Comment:
This should be conditional on OPENTHREAD_CONFIG_ENABLE_TOBLE as well.

Reply:
Ok.  Though what is the use case for leaving advertising on while a WoBLE connection is active?  BTP doesn't support multi-connection and closing the connection should restart WoBLE advertisements.

Reply:
This API instructs the connectivity manager to enable/disable advertising altogether.  It's value is entirely under the control of the application; it is never changed by the Device Layer.   So while BLEAdvertisingEnabled() == false, closing the connection will not restart advertising.

I suggest the following approach to coordinating advertising when ToBLE is enabled:

1) Add a compile-time option to the Device Layer that suppresses advertising while a WoBLE connection is active.  E.g.: WEAVE_DEVICE_CONFIG_SINGLE_WOBLE_CONNECTION.  For devices that use ToBLE, they would also need to enable this option.

2) In the example app, inside the event handler function, monitor all changes to the device's network, fabric and service provisioning states.  For any change to any of these states, evaluate whether the device is fully provisioned (network, fabric and service).  If so, disable advertising by calling SetBLEAdvertisingEnabled(false).  If not, enable advertising by calling SetBLEAdvertisingEnabled(true).

Using this approach, advertising is automatically disabled when a WoBLE connection is established, and only re-enabled when the connection ends and the device has not been fully provisioned.

Reply:
Ok, applied these suggestion in d178b9b.